Userland/Services/LaunchServer/Launcher.cpp

```diff
@@ -312,6 +312,19 @@ bool Launcher::open_file_url(const URL& url)
     String extension = {};
     if (extension_parts.size() > 1)
         extension = extension_parts.last();
-    return open_with_user_preferences(m_file_handlers, extension, url.path(), "/bin/TextEditor");
+
+    // Additionnal parameters parsing, specific for the file protocol
+    Vector<String> additional_parameters;
+    additional_parameters.append(url.path());
+    auto parameters = url.query().split('&');
+    for (auto parameter = parameters.begin(); parameter != parameters.end(); ++parameter) {
+        auto pair = parameter->split('=');
+        if (pair[0] == "line_number") {
```

Comment:
```suggestion
        if (pair.size() == 2 && pair[0] == "line_number") {
```

otherwise this would crash with something like "?line_number="